### DISCUSSION TITLE
Leaky ChromeDriver Processes

### DISCUSSION THREAD

**Discussion by jcanedo279** [ultrafunkamsterdam/undetected-chromedriver#1821] [General]
This is somewhat of a duplicate of #1613 with added observations. For context I am running UC on Windows 11, other platforms have a different experience, and using Chrome 123.0.6312.105, specifically CfT from https://googlechromelabs.github.io/chrome-for-testing/. On MacOS it seems that leaky processes are much harder to repro, which suggests there's something specific regarding Windows (at least 11) which keeps these processes running.

This comment seems to hit the nail on the head: https://github.com/ultrafunkamsterdam/undetected-chromedriver/discussions/1613#discussioncomment-7350474, @ongbanbanhbao suggests using the --disable-gpu option which seems to stop the process leak, the only issue being that this is pretty detectable on most sites, likely due to its association with headless browsing. The fact that this only happens on Windows suggests that there is something about the way UC utilizes the gpu (or configures/overrides the Selenium options/driver) which keeps these processes running.

I've tried disabling subprocesses and unsetting no_sandbox but those don't actually seem to stop leaky processes.

Does anyone understand why --disable-gpu is required to stop leaks when closing/quitting drivers? It seems a bit over the top to use the os to kill processes programmatically.
Does anyone know of a workaround which is less detectable?

**Comment by terra-alex** [comment]
same issue here.. the current workaround I've got is just a script that monitors system memory usage and restarts my docker container when the memory threshold exceeds 80%

**Reply by ultrafunkamsterdam** [reply]
This is not relevant to OP. You're using Docker

**Reply by Connor9994** [reply]
Even more damning because it has massive leaks even when containerized.

**Comment by Connor9994** [comment]
20GB of RAM in usage, none of which is visible in the Resource Monitor..

![13](https://github.com/user-attachments/assets/092a8314-a4e6-4c8b-bc3a-0642b57a9c75)

**Reply by ultrafunkamsterdam** [reply]
I don't see your python process. Sort it by usage.

**Reply by Connor9994** [reply]
This IS sorted by usage. There is no python running anymore. Application is completely closed.
All our script does is go to the site, sleep 3 seconds, close the driver. And we get >3GB of leak an hour. 

This has existed since/before .32

**Reply by ultrafunkamsterdam** [reply]
I think should be aware of what you are looking at. "Used" means something else then what one would think.
Used just means not "cleaned" up (which is due to lots of physical memory available so kernel has no need to yet).

Also

As soon as a process ends it can't keep memory locked or something.
